.github/workflows/deploy-to-gcp.yaml

```diff
@@ -1,16 +1,20 @@
 name: Deploy Xplorers API to GCP
-
-on: push
+on: 
+  push:
+  pull_request:
```

Comment:
does this restrict the workflow to pull_requests only? Will this work when we merge to `main`?

And should we be scoping this to just when a Pull Request is ready for review?

```
on:
  pull_request:
    types:
      - ready_for_review
```

Reply:
This should do the trick as its looking for a push event in branch main. However Suhail dai has suggested that we do the deployment (Terraform Apply) manually when tthe TF files are approved and someone do it.
So doing some modifications.